packages/core/src/util/renderer.ts

```diff
@@ -137,33 +138,65 @@ export const showAsRequired = (
 };
 
 /**
- * Create a default value based on the given scheam.
+ * Create a default value based on the given schema.
  * @param schema the schema for which to create a default value.
  * @returns {any}
  */
-export const createDefaultValue = (schema: JsonSchema) => {
-  switch (schema.type) {
-    case 'string':
-      if (
-        schema.format === 'date-time' ||
-        schema.format === 'date' ||
-        schema.format === 'time'
-      ) {
-        return new Date();
+export const createDefaultValue = (
+  schema: JsonSchema,
+  rootSchema: JsonSchema
+) => {
+  const resolvedSchema = Resolve.schema(schema, schema.$ref, rootSchema);
+  if (resolvedSchema.default !== undefined) {
+    return extractDefaults(resolvedSchema, rootSchema);
+  }
+  if (hasType(resolvedSchema, 'string')) {
+    if (
+      resolvedSchema.format === 'date-time' ||
+      resolvedSchema.format === 'date' ||
+      resolvedSchema.format === 'time'
+    ) {
+      return convertDateToString(new Date(), resolvedSchema.format);
+    }
+    return '';
+  } else if (
+    hasType(resolvedSchema, 'integer') ||
+    hasType(resolvedSchema, 'number')
+  ) {
+    return 0;
+  } else if (hasType(resolvedSchema, 'boolean')) {
+    return false;
+  } else if (hasType(resolvedSchema, 'array')) {
+    return [];
+  } else if (hasType(resolvedSchema, 'null')) {
+    return null;
+  } else if (hasType(resolvedSchema, 'object')) {
+    return extractDefaults(resolvedSchema, rootSchema);
+  } else {
```

Comment:
I would prefer checking first for object and then later for null. So in case people modeled nullable objects we are more constructive.